jobs/kola-azure.Jenkinsfile

```diff
@@ -0,0 +1,179 @@
+def pipeutils, streams, official
+node {
+    checkout scm
+    pipeutils = load("utils.groovy")
+    streams = load("streams.groovy")
+    def pipecfg = pipeutils.load_config()
+    azure_testing_resource_group = pipecfg['azure-testing-resource-group']
+    azure_testing_storage_account = pipecfg['azure-testing-storage-account']
+    azure_testing_storage_container = pipecfg['azure-testing-storage-container']
+    official = pipeutils.isOfficial()
+}
+
+properties([
+    pipelineTriggers([]),
+    parameters([
+      choice(name: 'STREAM',
+             // list devel first so that it's the default choice
+             choices: (streams.development + streams.production + streams.mechanical),
+             description: 'Fedora CoreOS stream to test'),
+      string(name: 'VERSION',
+             description: 'Fedora CoreOS Build ID to test',
+             defaultValue: '',
+             trim: true),
+      string(name: 'ARCH',
+             description: 'Target architecture',
+             defaultValue: 'x86_64',
+             trim: true),
+      string(name: 'S3_STREAM_DIR',
+             description: 'Override the Fedora CoreOS S3 stream directory',
+             defaultValue: '',
+             trim: true),
+      string(name: 'KOLA_TESTS',
+             description: 'Override tests to run',
+             defaultValue: "",
+             trim: true),
+      string(name: 'COREOS_ASSEMBLER_IMAGE',
+             description: 'Override the coreos-assembler image to use',
+             defaultValue: "coreos-assembler:main",
+             trim: true),
+      string(name: 'FCOS_CONFIG_COMMIT',
+             description: 'The exact config repo git commit to run tests against',
+             defaultValue: '',
+             trim: true),
+    ]),
+    buildDiscarder(logRotator(
+        numToKeepStr: '100',
+        artifactNumToKeepStr: '100'
+    )),
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+])
+
+// Locking so we don't run multiple times on the same region. We are speeding up our testing by dividing
+// the load to different regions depending on the architecture. This allows us not to exceed our quota in 
+// a single region while still being able to execute two test runs in parallel.
+lock(resource: "kola-azure-${params.ARCH}") {
+
+    currentBuild.description = "[${params.STREAM}][${params.ARCH}] - ${params.VERSION}"
+
+    // Use eastus region for now
+    def region = "eastus"
+
+    def s3_stream_dir = params.S3_STREAM_DIR
+    if (s3_stream_dir == "") {
+        s3_stream_dir = "fcos-builds/prod/streams/${params.STREAM}"
+    }
```

Comment:
Aside: I think with the removal of developer mode and officializing the requirement for a separate S3 bucket, I think we should be able to get rid of `S3_STREAM_DIR`.

Reply:
maybe in a followup..

Reply:
Oh yeah, didn't mean for this to be cleaned up in this PR.